MessagingService.EmailMessageAggregate/EmailAggregate.cs

```diff
@@ -200,41 +345,52 @@ private void PlayEvent(ResponseReceivedFromProviderEvent domainEvent)
         {
             this.ProviderEmailReference = domainEvent.ProviderEmailReference;
             this.ProviderRequestReference = domainEvent.ProviderRequestReference;
+            this.MessageStatus = MessageStatus.Sent;
         }
 
-        #endregion
-    }
-
-    /// <summary>
-    /// 
-    /// </summary>
-    internal class MessageRecipient
-    {
-        #region Constructors
-
-        #endregion
-
-        #region Properties
+        /// <summary>
+        /// Plays the event.
+        /// </summary>
+        /// <param name="domainEvent">The domain event.</param>
+        private void PlayEvent(MessageDeliveredEvent domainEvent)
+        {
+            this.MessageStatus = MessageStatus.Delivered;
+        }
 
         /// <summary>
-        /// Converts to address.
+        /// Plays the event.
         /// </summary>
-        /// <value>
-        /// To address.
-        /// </value>
-        internal String ToAddress { get; private set; }
+        /// <param name="domainEvent">The domain event.</param>
+        private void PlayEvent(MessageFailedEvent domainEvent)
+        {
+            this.MessageStatus = MessageStatus.Failed;
+        }
 
-        #endregion
+        /// <summary>
+        /// Plays the event.
+        /// </summary>
+        /// <param name="domainEvent">The domain event.</param>
+        private void PlayEvent(MessageRejectedEvent domainEvent)
+        {
+            this.MessageStatus = MessageStatus.Rejected;
+        }
 
-        #region Methods
+        /// <summary>
+        /// Plays the event.
+        /// </summary>
+        /// <param name="domainEvent">The domain event.</param>
+        private void PlayEvent(MessageBouncedEvent domainEvent)
+        {
+            this.MessageStatus = MessageStatus.Bounced;
+        }
 
         /// <summary>
-        /// Creates the specified to address.
+        /// Plays the event.
         /// </summary>
-        /// <param name="toAddress">To address.</param>
-        internal void Create(String toAddress)
+        /// <param name="domainEvent">The domain event.</param>
+        private void PlayEvent(MessageMarkedAsSpamEvent domainEvent)
```

Comment:
Codacy found an issue: [Remove this unused method parameter 'domainEvent'.](https://app.codacy.com/manual/stuart_ferguson1/Messaging/pullRequest?prid=5953475)